src/accounts/views.py

```diff
@@ -1,31 +1,92 @@
 # Create your views here.
-from django.contrib.auth import login
+import os
+from django.contrib.auth import login, logout
 from django.shortcuts import render, redirect
 from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.contrib.auth.models import User
+from django.contrib.messages import get_messages
+from django.db.models.signals import post_migrate
+from django.dispatch import receiver
+from allauth.socialaccount.models import SocialApp
+from dotenv import load_dotenv, set_key
 from .forms import MyAuthenticationForm, MyUserCreationForm
 
+# In your app's views.py or signals.py
+@receiver(post_migrate)
+def my_signal_receiver(sender, **kwargs): # pylint: disable=unused-argument
+    # Safe to access the database here
+
+    # Create Site
+    from django.contrib.sites.models import Site
+
+    site, _ = Site.objects.get_or_create(
+        domain=os.getenv('domain','http://127.0.0.1:8000'),
+        defaults={'name': os.getenv('domain_name', 'http://127.0.0.1:8000')}
+    )
+
+    site.save()
+
+    # Path to your .env file
+    load_dotenv()
+    env_file = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), '.env') # pylint: disable=locally-disabled, multiple-statements, fixme, line-too-long
+    set_key(env_file, 'SITE_ID', str(site.id)) # Write or update SITE_ID
+
+    # create SocialAPP
+    app, _ = SocialApp.objects.get_or_create(provider='google', name='google')
+    app.client_id = os.getenv('client_id', 'none')
+    app.secret = os.getenv('secret_key', 'none')
+    app.sites.add(site)
+    app.save()
+
 def register_view(request):
     if request.method == 'POST':
         form = MyUserCreationForm(request.POST)
-        if form.is_valid():
-            user = form.save()
-            login(request, user)
-            return redirect('profile')
+        email = request.POST.get('email')
+
+        # Check if the email is already in use
+        if User.objects.filter(email=email).exists():
+            messages.error(request, "This email is already registered.")
+            return render(request, 'accounts/register.html', {'form': form})
+
+        if form.is_valid(): # pylint: disable=no-else-return, locally-disabled, multiple-statements, fixme, line-too-long
+            user = form.save(commit=False)  # Create a user object but don't save it yet
+            user.username = form.cleaned_data.get('email')  # Set username as email
+            user.save()  # Save the user object
+            login(request, user, backend='django.contrib.auth.backends.ModelBackend')
+            return redirect('accounts:profile')
+        else:
+            messages.error(request, "Registration Failed!! Password mismatch or too generic") # pylint: disable=locally-disabled, multiple-statements, fixme, line-too-long
```

Comment:
I don't think you need any pylint disable here. There's a trailing space at the end of the line you can get rid of and you can split this into two lines.

```python
            messages.error(request,
                           "Registration Failed!! Password mismatch or too generic") 
```